GoogleChromeLabs/ps-analysis-tool
# Feature: Export report in an HTML file

## Description

This PR adds the capability to export an HTML file to the extension. The HTML file will contain data about 1st and 3rd party cookies blocked cookies and frames.

## Relevant Technical Choices
- Added a new view in `packages/extension/src/view` named `report`.
- The report view is a simple react SPA which reads data from `window.PSAT_DATA`
- The report view is built alongside all other views and marked as a web accessible resource for the extension.
- While exporting this html file is fetched and `window.PSAT_DATA` is injected then the html file is downloaded.



## Testing Instructions
- Click the 'Download Button' in the cookie dashboard.
- A file named `report.html` file will be downloaded.
- The file can be opened in any browser the UI should be similar to the cookie dashboard.



## Additional Information:

### Todo
- [x] (Bug) Extra item, Frame no and associated frames visible in report. 
- [x] Add Library detection data in the report. 
- [x] Disable download button till library detection data is ready.
  - [x] Refactor Button component to have a disabled state. ( Done in PR #585 ) 
  - [x] Use Library detection provider to disable download report button.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/53055971/e34a33bb-8966-4d1c-8019-25e2d38d4051




---

## Checklist






Partially Addresses #228
